File: output/gpt_4o/avatar/direct_translation/python/java/failed_translation/codeforces_30_A.java

import java.util.Scanner;

public class codeforces_30_A {
    public static int task4(int A, int B, int n) {
        for (int X = -1000; X <= 1000; X++) {
            if (A * Math.pow(X, n) == B) {
                return X;
            }
        }
        return Integer.MAX_VALUE; // Represents "No solution"
    }

    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        String[] input = scanner.nextLine().split(" ");
        int A = Integer.parseInt(input[0]);
        int B = Integer.parseInt(input[1]);
        int n = Integer.parseInt(input[2]);
        int result = task4(A, B, n);
        if (result == Integer.MAX_VALUE) {
            System.out.println("No solution");
        } else {
            System.out.println(result);
        }
        scanner.close();
    }
}

File: output/gpt_4o/avatar/direct_translation/java/python/codeforces_30_A.py

import math

def solve():
    sc = map(int, input().split())
    A, B, n = sc
    if A == 0:
        if B == 0:
            print(1)
        else:
            print("No solution")
        return
    if B % A != 0:
        print("No solution")
        return
    B //= A
    neg = B < 0
    if neg and n % 2 == 0:
        print("No solution")
        return
    if neg:
        B = -B
    for x in range(B + 1):
        if math.pow(x, n) == B:
            print(-x if neg else x)
            return
    print("No solution")

solve()

